### ISSUE TITLE
Multiple velero + multiple minio instances in separate namespaces work but restic partially fails (due to different cloud credentials)

### ISSUE THREAD

**Issue by mierea** [vmware-tanzu/velero#7162]
**What steps did you take and what happened:**

- deploy 3x minio in different kubernetes namespaces (mf-realtime, mf-scheduled, mf-additional)
- deploy 3x velero in different kubernetes namespaces(mf-realtime, mf-scheduled, mf-additional)
- mf-realtime and mf-scheduled re-use same user and password for cloud credentials
- mf-additional has different credential set

**What did you expect to happen:**

I was expecting the backups to work well on all 3 instances. On the mf-additional namespace the restic backup is reported to have partially failed (although in minio I cleary see the backup being there and I can also create a restore from it)

**The following information will help us better understand what's going on**:

_If you are using velero v1.7.0+:_  
Please use `velero debug  --backup <backupname> --restore <restorename>` to generate the support bundle, and attach to this issue, more options please refer to `velero debug --help` 
[bundle-2023-11-29-07-19-08.tar.gz](https://github.com/vmware-tanzu/velero/files/13497372/bundle-2023-11-29-07-19-08.tar.gz)

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by allenxu404** [comment]
I found the follow error in the given bundle: 

```
error="pod volume backup failed: error creating uploader: failed to connect repository: error running command=restic snapshots --repo=s3:http://minio-mf-additional.minio-mf-additional.svc.cluster.local:9000/velero-mf-additional/restic/daily-boss --password-file=/tmp/credentials/velero/velero-repo-credentials-repository-password --cache-dir=/scratch/.cache/restic --latest=1, stdout=, stderr=Fatal: unable to open config file: Stat: The Access Key Id you provided does not exist in our records.\nIs there a repository at the following location?\ns3:http://minio-mf-additional.minio-mf-additional.svc.cluster.local:9000/velero-mf-additional/restic/daily-boss\n: exit status 1"
```

The root cause seems to be:
```
stderr=Fatal: unable to open config file: Stat: The Access Key Id you provided does not exist in our records.
Is there a repository at the following location?
s3:http://minio-mf-additional.minio-mf-additional.svc.cluster.local:9000/velero-mf-additional/restic/daily-boss
: exit status 1" 
```

Please double check the s3 url and ensure the the credentials file used to create the Secret object is well formatted and access key id is correct.

**Comment by allenxu404** [comment]
Can I know which Velero version you're using?

**Comment by allenxu404** [comment]
I also saw a few errors below in the given bundle: 
```
dial tcp: lookup minio-mf-additional.minio-mf-additional.svc.cluster.local on 127.0.0.53:53: server misbehaving
```

It seems the errors may have been caused by DNS issues, Can you look into it?

**Comment by mierea** [comment]
> I also saw a few errors below in the given bundle:
> 
> ```
> dial tcp: lookup minio-mf-additional.minio-mf-additional.svc.cluster.local on 127.0.0.53:53: server misbehaving
> ```
> 
> It seems the errors may have been caused by DNS issues, Can you look into it?

Doesn't seem to be an issue because  as soon as I redeploy using the cloud credentials used by the other instances everything works and I no longer receive a false-positive on PartiallyFailed backups (restic side of things)

Also the error you have mentiond I have on all velero instances.

Example of the same error on an velero-mf-realtime:
```
dial tcp: lookup minio-mf-realtime.minio-mf-realtime.svc.cluster.local on 127.0.0.53:53: server misbehaving>
```
And the backup from velero-mf-realtime:
```
NAME           STATUS      ERRORS   WARNINGS   CREATED                         EXPIRES   STORAGE LOCATION   SELECTOR
daily-boss-1   Completed   0        0          2023-11-29 06:59:49 +0000 UTC   23d       default            <none>
```

And the partially failed from velero-mf-additional:
```
NAME           STATUS            ERRORS   WARNINGS   CREATED                         EXPIRES   STORAGE LOCATION   SELECTOR
daily-boss-1   PartiallyFailed   1        0          2023-11-29 07:17:26 +0000 UTC   23d       default            <none>
```

**Comment by allenxu404** [comment]
I will set up a test env to attempt reproducing this issue, and I expect the multiple BSLs not to interfere with each other.

**Comment by allenxu404** [comment]
Under the further review of the data in the given bundle, I found the process of the backup  `daily-boss-1` in the namespace `velero-mf-additional` proceeded as follows:

Node agent controller initially failed to backup the volume with the following error (which seemed like an environment issue rather than Velero issue) and this error was also captured in PodVolumeBackup's status:
```
time="2023-11-29T07:17:30Z" level=error msg="Error backing up item" backup=velero-mf-additional/daily-boss-1 error="pod volume backup failed: error creating uploader: failed to connect repository: error running command=restic snapshots --repo=s3:http://minio-mf-additional.minio-mf-additional.svc.cluster.local:9000/velero-mf-additional/restic/daily-boss --password-file=/tmp/credentials/velero/velero-repo-credentials-repository-password --cache-dir=/scratch/.cache/restic --latest=1, stdout=, stderr=Fatal: unable to open config file: Stat: The Access Key Id you provided does not exist in our records.\nIs there a repository at the following location?\ns3:http://minio-mf-additional.minio-mf-additional.svc.cluster.local:9000/velero-mf-additional/restic/daily-boss\n: exit status 1" error.file="/go/src/github.com/vmware-tanzu/velero/pkg/podvolume/backupper.go:324" error.function="github.com/vmware-tanzu/velero/pkg/podvolume.(*backupper).BackupPodVolumes" logSource="pkg/backup/backup.go:448" name=boss-web-deployment-864f99dc66-g8jhf
``` 

Node agent controller took a second attempt and succeed:
```
time="2023-11-29T07:18:04Z" level=info msg="Async fs backup data path completed" PVB=daily-boss-1-4zllj controller=PodVolumeBackup logSource="pkg/controller/pod_volume_backup_controller.go:195" pvb=daily-boss-1-4zllj
time="2023-11-29T07:18:04Z" level=info msg="PodVolumeBackup completed" controller=PodVolumeBackup logSource="pkg/controller/pod_volume_backup_controller.go:225" pvb=daily-boss-1-4zllj
```

We can also see this success from the PodVolumeBackup status:
```
   "status": {
                "completionTimestamp": "2023-11-29T07:18:04Z",
                "message": "error creating uploader: failed to connect repository: error running command=restic snapshots --repo=s3:http://minio-mf-additional.minio-mf-additional.svc.cluster.local:9000/velero-mf-additional/restic/daily-boss --password-file=/tmp/credentials/velero/velero-repo-credentials-repository-password --cache-dir=/scratch/.cache/restic --latest=1, stdout=, stderr=Fatal: unable to open config file: Stat: The Access Key Id you provided does not exist in our records.\nIs there a repository at the following location?\ns3:http://minio-mf-additional.minio-mf-additional.svc.cluster.local:9000/velero-mf-additional/restic/daily-boss\n: exit status 1",
                "path": "/host_pods/8cb6d6cd-1dd4-46d2-a19f-ed32e75e1f9d/volumes/kubernetes.io~csi/pvc-8cad1829-4671-4380-8dcc-b1d8843c2e4a/mount",
                "phase": "Completed",
                "progress": {
                    "bytesDone": 1187892943,
                    "totalBytes": 1187892943
                },
                "snapshotID": "bc7813a6",
                "startTimestamp": "2023-11-29T07:17:30Z"
            }
```

However,  Velero controller only detected the first result so it marked the status of backup `daily-boss-1` as `PartiallyFailed`.  
We can check the related logic here: https://github.com/vmware-tanzu/velero/blob/06d12dec4030703cb46b2699ca434cdabd9edbb0/pkg/podvolume/backupper.go#L323 

In summary,  the backup status was incorrectly reported as "PartiallyFailed" rather than "Completed",  even though the volume backup was ultimately successful.

**Comment by allenxu404** [comment]
Under the further review of the data in the given bundle, I found the PodVolumeBackup `daily-boss-1-4zllj` which references the backup `daily-boss-1` in the namespace `velero-mf-additional` was unexpectedly processed twice by different node agent in different namespace. 

The mismatched node agent detected that PodVolumeBackup and failed to back up the volume due to the mismatched credentials with the following error and this error was also recorded in the PodVolumeBackup's '.status.message' field: 
```
failed to connect repository: error running command=restic snapshots --repo=s3:http://minio-mf-additional.minio-mf-additional.svc.cluster.local:9000/velero-mf-additional/restic/daily-boss --password-file=/tmp/credentials/velero/velero-repo-credentials-repository-password --cache-dir=/scratch/.cache/restic --latest=1, stdout=, stderr=Fatal: unable to open config file: Stat: The Access Key Id you provided does not exist in our records.\
```

Concurrently, the matched node agent in the namespace `velero-mf-additional` also detected that PodVolumeBackup and successfully backed up the volume. The following log messages from the node agent provided evidence on that:
```
time="2023-11-29T07:18:04Z" level=info msg="Async fs backup data path completed" PVB=daily-boss-1-4zllj controller=PodVolumeBackup logSource="pkg/controller/pod_volume_backup_controller.go:195" pvb=daily-boss-1-4zllj
time="2023-11-29T07:18:04Z" level=info msg="PodVolumeBackup completed" controller=PodVolumeBackup logSource="pkg/controller/pod_volume_backup_controller.go:225" pvb=daily-boss-1-4zllj
```
The final PodVolumeBackup status is displayed below, we can see that the final phase is marked as `Completed` but it still included the error message from the first processing:
```
 "status": {
                "completionTimestamp": "2023-11-29T07:18:04Z",
                "message": "error creating uploader: failed to connect repository: error running command=restic snapshots --repo=s3:http://minio-mf-additional.minio-mf-additional.svc.cluster.local:9000/velero-mf-additional/restic/daily-boss --password-file=/tmp/credentials/velero/velero-repo-credentials-repository-password --cache-dir=/scratch/.cache/restic --latest=1, stdout=, stderr=Fatal: unable to open config file: Stat: The Access Key Id you provided does not exist in our records.\nIs there a repository at the following location?\ns3:http://minio-mf-additional.minio-mf-additional.svc.cluster.local:9000/velero-mf-additional/restic/daily-boss\n: exit status 1",
                "path": "/host_pods/8cb6d6cd-1dd4-46d2-a19f-ed32e75e1f9d/volumes/kubernetes.io~csi/pvc-8cad1829-4671-4380-8dcc-b1d8843c2e4a/mount",
                "phase": "Completed",
                "progress": {
                    "bytesDone": 1187892943,
                    "totalBytes": 1187892943
                },
                "snapshotID": "bc7813a6",
                "startTimestamp": "2023-11-29T07:17:30Z"
            }

```

Velero only detected the result from the first processing, so the backup status  was reported as `PartiallyFailed` rather than `Completed`.

Just curious in what scenario you have to set up multiple Velero within one cluster?  Seems that it's not a good practice to use velero that way.

**Comment by mierea** [comment]
Hi @allenxu404 

We have multiple teams working on same product but their components have different lifecycles and policies and require different destinations for the backups.

**Comment by mierea** [comment]
Hi @allenxu404 ,

velero client is 
```
Client:
        Version: v1.2.0
        Git commit: 5d008491bbf681658d3e372da1a9d3a21ca4c03c
```

`velero namespace: velero/velero:v1.10.1` is an older velero deploy on same kubernetes cluster.
Also the problem is the same when directly scheduling backups with kubectl. (so velero client is not used at all in this case)

**Comment by allenxu404** [comment]
Can you upgrade the client and ensure Velero version remains at v1.12.2 both on server and client side? 

It appears that the node agent restricts the cache to its own namespace, which might prevent PodVolumeBackup from being processed by multiple node agents.
https://github.com/vmware-tanzu/velero/blob/06d12dec4030703cb46b2699ca434cdabd9edbb0/pkg/cmd/cli/nodeagent/server.go#L174
